air/src/execution_step/lambda_applier/applier.rs

```diff
@@ -104,13 +111,62 @@ fn select_by_path_from_stream<'value>(
     Ok(select_result)
 }
 
+fn select_by_path_from_stream_<'value>(
+    stream: impl ExactSizeIterator<Item = (&'value JValue, RcSecurityTetraplet)> + 'value,
+    lambda: &NonEmpty<ValueAccessor<'_>>,
+    exec_ctx: &ExecutionCtx<'_>,
+) -> ExecutionResult<MapLensResult<'value>> {
+    let (prefix, body) = lambda.split_first();
+    let idx = match prefix {
+        ValueAccessor::ArrayAccess { idx } => *idx,
+        ValueAccessor::FieldAccessByName { field_name } => {
+            return lambda_to_execution_error!(Err(LambdaError::FieldAccessorAppliedToStream {
+                field_name: field_name.to_string(),
+            }));
+        }
+        ValueAccessor::FieldAccessByScalar { scalar_name } => {
+            let scalar = exec_ctx.scalars.get_value(scalar_name)?;
+            lambda_to_execution_error!(try_scalar_ref_as_idx(scalar))?
+        }
+        ValueAccessor::Error => unreachable!("should not execute if parsing succeeded. QED."),
+    };
+    let idx = idx as usize;
+    let stream_size = stream.len();
+
+    let (value, tetraplet) = lambda_to_execution_error!(stream
+        .peekable()
+        .nth(idx)
+        .ok_or(LambdaError::CanonStreamNotHaveEnoughValues { stream_size, idx }))?;
+
+    // 3d case take tetraplet from this VA and apply the full lens into it.
+    let select_result = if body.is_empty() {
+        let result = Cow::Borrowed(value);
+        MapLensResult::from_cow(result, tetraplet)
+    } else {
+        let SecurityTetraplet {
+            peer_pk,
+            service_id,
+            function_name,
+            json_path,
+        } = tetraplet.as_ref();
+
+        let json_path_suffix = body.iter().fold("".to_string(), |acc, va| acc + &va.to_string());
+        let json_path = json_path.to_string() + &prefix.to_string() + &json_path_suffix;
```

Comment:
you can use `add_lambda` method

Reply:
But I create a new tetraplet using the constructed path. Should I use add_lambda() in this case?